q2_longitudinal/plugin_setup.py

```diff
@@ -29,14 +35,84 @@
 )
 
 
+# FirstDifferencesFormat is a near replica of AlphaDiversityFormat in q2_types
+FirstDifferences = SemanticType(
+    'FirstDifferences', variant_of=SampleData.field['type'])
+
+
+class FirstDifferencesFormat(model.TextFileFormat):
+    def _check_n_records(self, n_lines=None):
+        with self.open() as fh:
+            line = fh.readline()
+            for line, _ in zip(fh, range(n_lines)):
+                cells = line.strip().split('\t')
+                if len(cells) != 2:
+                    raise ValidationError('File is not FirstDifferencesFormat')
+
+    def _validate_(self, level):
+        record_count_map = {'min': 5, 'max': 1000}
+        self._check_n_records(record_count_map[level])
+
+
+FirstDifferencesDirectoryFormat = model.SingleFileDirectoryFormat(
+    'FirstDifferencesDirectoryFormat', 'FirstDifferences.tsv',
+    FirstDifferencesFormat)
+
+
+# borrowed from q2_types
+def _read_dataframe(fh):
+    # Using `dtype=object` and `set_index` to avoid type casting/inference
+    # of any columns or the index.
+    df = pd.read_csv(fh, sep='\t', header=0, dtype=object)
+    df.set_index(df.columns[0], drop=True, append=False, inplace=True)
+    df.index.name = None
+    return df
+
+
+@plugin.register_transformer
+def _4(data: pd.Series) -> (FirstDifferencesFormat):
+    ff = FirstDifferencesFormat()
+    with ff.open() as fh:
+        data.to_csv(fh, sep='\t', header=True)
+    return ff
+
+
+@plugin.register_transformer
+def _5(ff: FirstDifferencesFormat) -> (pd.Series):
```

Comment:
For this transformer and the Metadata transformer below, can you cast the column to float so that the differences aren't a string dtype in the object?

Reply:
👍